Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -225,4 +306,71 @@ private void OnRequestWithdraw(EntityUid uid, StoreComponent component, StoreReq
         component.Balance[msg.Currency] -= msg.Amount;
         UpdateUserInterface(buyer, uid, component);
     }
+
+    private void OnRequestRefund(EntityUid uid, StoreComponent component, StoreRequestRefundMessage args)
+    {
+        // TODO: Remove guardian/holopara
+
+        if (args.Session.AttachedEntity is not { Valid: true } buyer)
+            return;
+
+        if (!IsOnStartingMap(uid, component))
+        {
+            component.RefundAllowed = false;
+            UpdateUserInterface(buyer, uid, component);
+        }
+
+        if (!component.RefundAllowed || !component.BoughtEntities.Any())
+            return;
+
+        foreach (var purchase in component.BoughtEntities.ToList())
+        {
+            if (purchase.Valid)
+            {
+                component.BoughtEntities.Remove(purchase);
+
+                if (_actions.TryGetActionData(purchase, out var actionComponent))
+                {
+                    _actionContainer.RemoveAction(purchase, actionComponent);
+                    EntityManager.DeleteEntity(purchase);
+                }
+                else
+                {
+                    EntityManager.DeleteEntity(purchase);
+                }
+            }
```

Comment:
You're always deleting entity so you can remove 1 call here and put it outta the scope.